astropy/units/quantity.py

```diff
@@ -373,7 +373,7 @@ def __new__(cls, value, unit=None, dtype=None, copy=True, order=None,
                     copy = False  # copy will be made in conversion at end
 
         value = np.array(value, dtype=dtype, copy=copy, order=order,
-                         subok=False, ndmin=ndmin)
+                         subok=True, ndmin=ndmin)
```

Comment:
Note: this could also be `subok=subok` so that `Quantity` initialization itself always returns a pure `Quantity` - on the other hand, it is not clear that getting a Quantity with a structured dtype out of a Distribution is useful for anything.

But I can change this if requested - most important I feel is that `distribution * unit` gives a `QuantityDistribution`.

Reply:
Does this potentially open a problem if doing NDData * unit?

Reply:
Since `NDData` is not an `ndarray` subclass, it will likely break regardless... (though perhaps `nddata` overrides `__mul__` - it is one piece of astropy I have not really interacted with...).

Reply:
It doesn't appear to. Just had a quick peak.

Reply:
That might be a good PR though... ``NDData * unit = NDData(..., unit=old_unit * unit)``